Structural/Decorator/JsonRenderer.php

```diff
@@ -8,4 +8,16 @@ public function renderData(): string
     {
         return json_encode($this->wrapped->renderData());
     }
+
+    /**
+     * Renders data as JSON based on options passed.
+     * In terms of decorator patters, it represents additional extended behaviour.
+     *
+     * @param int $options Options to be used to encode data into JSON.
+     * @return string Rendered output
+     */
+    public function renderDataWithOptions($options = 0): string
```

Comment:
I like @adayth solution here (of course complete with type hints I would be even better). But to show chaining of decorators, this is a bad example and should be replaced with something new.